Framework/DataObjects/src/EventList.cpp

```diff
@@ -1946,6 +2009,44 @@ void EventList::generateHistogram(const MantidVec &X, MantidVec &Y, MantidVec &E
   }
 }
 
+// --------------------------------------------------------------------------
+/** Generates both the Y and E (error) histograms w.r.t TOF
+ * for an EventList with or without WeightedEvents.
+ *
+ * This calculates histogram without sorting the events by using the step size to estimate the bin number. This has been
+ * made to only work for logarithmic or linear binning. This falls back to using the sorted histogram method if the
+ * events are already sorted, as that will be faster.
+ *
+ * @param step: bin step size
+ * @param X: x-bins supplied
+ * @param Y: counts returned
+ * @param E: errors returned
+ * @param skipError: skip calculating the error. This has no effect for weighted
+ *        events; you can just ignore the returned E vector.
+ */
+void EventList::generateHistogram(const double step, const MantidVec &X, MantidVec &Y, MantidVec &E,
+                                  bool skipError) const {
+  // if events are already sorted, use faster sorted histogram method
+  if (isSortedByTof())
```

Comment:
Do we want to funnel histograming empty event lists through the "traditional route" too?
```suggestion
  if (isSortedByTof() || isEmpty())
```